wp-includes/rest-api/auth/class-wp-rest-key-pair.php

```diff
@@ -0,0 +1,798 @@
+						),
+						'api_secret' => array(
+							'description' => esc_html__( 'The hashed API secret.', 'jwt-auth' ),
+							'type'        => 'string',
+							'readonly'    => true,
+						),
+						'created'    => array(
+							'description' => esc_html__( 'The date the key-pair was created.', 'jwt-auth' ),
+							'type'        => 'string',
+							'readonly'    => true,
+						),
+						'last_used'  => array(
+							'description' => esc_html__( 'The last time the key-pair was used.', 'jwt-auth' ),
+							'type'        => 'string',
+							'readonly'    => true,
+						),
+						'last_ip'    => array(
+							'description' => esc_html__( 'The last IP address that used the key-pair.', 'jwt-auth' ),
+							'type'        => 'string',
+							'readonly'    => true,
+						),
+					),
+				),
+			),
+		);
+
+		/**
+		 * Filters the REST endpoint schema.
+		 *
+		 * @param string $schema The endpoint schema.
+		 */
+		return apply_filters( 'rest_authentication_key_pair_schema', $schema );
+	}
+
+	/**
+	 * Display the key-pair section in a users profile.
+	 *
+	 * @since 0.1
+	 *
+	 * @param WP_User $user WP_User object of the logged-in user.
+	 */
+	public function show_user_profile( WP_User $user ) {
+		wp_enqueue_style( 'key-pair-css', JWT_AUTH_PLUGIN_URL . '/wp-admin/css/key-pair.css', array(), JWT_AUTH_VERSION );
+		wp_enqueue_script( 'key-pair-js', JWT_AUTH_PLUGIN_URL . '/wp-admin/js/key-pair.js', array(), JWT_AUTH_VERSION, true );
+		wp_localize_script(
+			'key-pair-js',
+			'keyPair',
+			array(
+				'nonce'   => wp_create_nonce( 'wp_rest' ),
+				'root'    => self::get_rest_uri(),
+				'token'   => WP_REST_Token::get_rest_uri(),
+				'user_id' => $user->ID,
+				'text'    => array(
+					/* translators: %s: key-pair name */
+					'confirm_one' => esc_html__( 'Revoke the %s key-pair? This action cannot be undone.', 'jwt-auth' ),
+					'confirm_all' => esc_html__( 'Revoke all key-pairs? This action cannot be undone.', 'jwt-auth' ),
+				),
+			)
+		);
+		$this->show_key_pair_section( $user );
+		$this->template_new_key_pair();
+		$this->template_new_token_key_pair();
+		$this->template_key_pair_row();
+	}
+
+	/**
+	 * Fires after the user's password is reset.
+	 *
+	 * @param WP_User $user The user.
+	 */
+	public function after_password_reset( WP_User $user ) {
+		if ( 'after_password_reset' !== current_filter() ) {
+			return;
+		}
+
+		$keypairs = $this->get_user_key_pairs( $user->ID );
+		if ( ! empty( $keypairs ) ) {
+			$this->set_user_key_pairs( $user->ID, array() );
+		}
+	}
+
+	/**
+	 * Fires after the user's password is reset.
+	 *
+	 * @param int $user_id The user ID.
+	 */
+	public function profile_update( $user_id ) {
+		if ( 'profile_update' !== current_filter() ) {
+			return;
+		}
+
+		if ( isset( $_POST['pass1'] ) && ! empty( $_POST['pass1'] ) ) { // phpcs:ignore
+			$keypairs = $this->get_user_key_pairs( $user_id );
+			if ( ! empty( $keypairs ) ) {
+				$this->set_user_key_pairs( $user_id, array() );
```

Comment:
Could use a clarifying comment, here or in the function docbloc, clearly explaining what's being cleared.